testsuite/MDAnalysisTests/coordinates/test_dcd.py

```diff
@@ -65,420 +82,287 @@ def test_with_statement(self):
-
-    def test_dt(self):
-        DT = 5.0
-        t = self.universe.trajectory
-        with self.Writer(self.outfile,
-                         t.n_atoms,
-                         dt=DT) as W:  # set time step to 5 ps
-            for ts in self.universe.trajectory:
-                W.write_next_timestep(ts)
-
-        uw = mda.Universe(PSF, self.outfile)
+        uw = mda.Universe(PSF, outfile)
         assert_almost_equal(uw.trajectory.totaltime,
-                            (uw.trajectory.n_frames - 1) * DT, 5)
+                            (uw.trajectory.n_frames - 1) * dt, decimal)
         times = np.array([uw.trajectory.time for ts in uw.trajectory])
         frames = np.array([ts.frame for ts in uw.trajectory])
-        assert_array_almost_equal(times, frames * DT, 5)
+        assert_array_almost_equal(times, frames * dt, decimal)
 
-    def test_OtherWriter(self):
-        t = self.universe.trajectory
-        W = t.OtherWriter(self.outfile)
-        for ts in self.universe.trajectory:
-            W.write_next_timestep(ts)
-        W.close()
 
-        uw = mda.Universe(PSF, self.outfile)
+@pytest.mark.parametrize("ext, decimal", (("dcd", 5),
+                                          ("xtc", 2)))
+def test_other_writer(universe_dcd, tmpdir, ext, decimal):
+    t = universe_dcd.trajectory
+    outfile = "test.{}".format(ext)
+    with tmpdir.as_cwd():
+        with t.OtherWriter(outfile) as W:
+            for ts in universe_dcd.trajectory:
+                W.write_next_timestep(ts)
 
+        uw = mda.Universe(PSF, outfile)
         # check that the coordinates are identical for each time step
-        for orig_ts, written_ts in zip(self.universe.trajectory,
+        for orig_ts, written_ts in zip(universe_dcd.trajectory,
                                        uw.trajectory):
-            assert_array_almost_equal(written_ts._pos, orig_ts._pos, 3,
+            assert_array_almost_equal(written_ts.positions, orig_ts.positions,
+                                      decimal,
                                       err_msg="coordinate mismatch between "
                                       "original and written trajectory at "
-                                      "frame %d (orig) vs %d (written)" % (
+                                      "frame {} (orig) vs {} (written)".format(
                                           orig_ts.frame, written_ts.frame))
 
-    def test_single_frame(self):
-        u = mda.Universe(PSF, CRD)
-        W = mda.Writer(self.outfile, u.atoms.n_atoms)
-        W.write(u.atoms)
-        W.close()
-        w = mda.Universe(PSF, self.outfile)
-        assert_equal(w.trajectory.n_frames, 1,
-                     "single frame trajectory has wrong number of frames")
-        assert_almost_equal(w.atoms.positions,
-                            u.atoms.positions,
-                            3,
-                            err_msg="coordinates do not match")
 
-    def test_with_statement(self):
```

Comment:
the writer is tested. The reader isn't. But I'm not aware we ever claimed that the normal reader supports a context manager protokoll